### PR TITLE
Overhaul with Pydantic and format version generic 0.3 / model 0.5

### PR DESCRIPTION
**Pull request by FynnBe**

### CODE REVIEW

**Comment by oeway** [comment]
FYI, I created an ImJoy plugin for react json schema form: https://oeway.github.io/imjoy-json-schema-form/ we can use it in Jupyter notebooks or the model zoo website. Try it at https://jupyter.imjoy.io Useful for generating model yaml, e.g. for the model builder.

**Comment by FynnBe** [comment]
> FYI, I created an ImJoy plugin for react json schema form: https://oeway.github.io/imjoy-json-schema-form/ we can use it in Jupyter notebooks or the model zoo website. Try it at https://jupyter.imjoy.io Useful for generating model yaml, e.g. for the model builder.

I tried it out: [test.ipynb.txt](https://github.com/bioimage-io/spec-bioimage-io/files/12352839/test.ipynb.txt)
But maybe the current json schema is simply too big... I get a white pop up windows only:
image

**Comment by oeway** [comment]
> > FYI, I created an ImJoy plugin for react json schema form: https://oeway.github.io/imjoy-json-schema-form/ we can use it in Jupyter notebooks or the model zoo website. Try it at https://jupyter.imjoy.io Useful for generating model yaml, e.g. for the model builder.
> 
> I tried it out: [test.ipynb.txt](https://github.com/bioimage-io/spec-bioimage-io/files/12352839/test.ipynb.txt) But maybe the current json schema is simply too big... edit: log is full of "Could not find a definition for #/$defs/bioimageio__spec__application__v0_2__Application." etc.. seems like the defs are not supported or there is some other compatiblity issue with the json schema? I get a white pop up window only: image

```
react-dom.production.min.js:189 Error: Could not find a definition for #/$defs/bioimageio__spec__application__v0_2__Application.
    at M (utils.js:614:9)
    at z (utils.js:693:20)
    at q (utils.js:681:12)
    at B (utils.js:704:24)
    at S (SchemaField.js:279:27)
    at t.value (SchemaField.js:433:14)
    at Ts (react-dom.production.min.js:204:189)
    at As (react-dom.production.min.js:203:149)
    at Ec (react-dom.production.min.js:292:172)
    at wu (react-dom.production.min.js:280:389)
ps @ react-dom.production.min.js:189
r.callback @ react-dom.production.min.js:189
qo @ react-dom.production.min.js:144
kc @ react-dom.production.min.js:262
wc @ react-dom.production.min.js:260
bc @ react-dom.production.min.js:259
(anonymous) @ react-dom.production.min.js:283
xu @ react-dom.production.min.js:281
iu @ react-dom.production.min.js:270
k @ scheduler.production.min.js:13
T @ scheduler.production.min.js:14
utils.js:614 Uncaught Error: Could not find a definition for #/$defs/bioimageio__spec__application__v0_2__Application.
    at M (utils.js:614:9)
    at z (utils.js:693:20)
    at q (utils.js:681:12)
    at B (utils.js:704:24)
    at S (SchemaField.js:279:27)
    at t.value (SchemaField.js:433:14)
    at Ts (react-dom.production.min.js:204:189)
    at As (react-dom.production.min.js:203:149)
    at Ec (react-dom.production.min.js:292:172)
    at wu (react-dom.production.min.js:280:389)
```

Removing all the `discriminator` eliminated  the error, however, it still complaining about the `root`. You can also test it with the official page here: https://rjsf-team.github.io/react-jsonschema-form/

**Comment by FynnBe** [comment]
> You can also test it with the official page here: https://rjsf-team.github.io/react-jsonschema-form/

there it works just fine
image

**Comment by FynnBe** [comment]
fixes https://github.com/conda-forge/bioimageio.spec-feedstock/issues/61

File: bioimageio/spec/_internal/io_utils.py [hunk 1]
```diff
@@ -272,3 +274,12 @@ def get_sha256(path: Path) -> Sha256:
     sha = h.hexdigest()
     assert len(sha) == 64
     return Sha256(sha)
+
+
+def load_array(source: FileSource) -> NDArray[Any]:
+    local_source = download(source).path
+    return numpy.load(local_source, allow_pickle=False)
```
Comment (oeway) [inline bioimageio/spec/_internal/io_utils.py:281]: Good that you added `allow_pickle=False`, could you also mention this in the spec docs, so we add a note on how they should save the array?

Maybe a todo for the bioimageio-core is to make sure we load it and export it with `allow_pickle=False` in the `build_model` function.

**Pull request merged**